akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MarshallingDirectivesSpec.scala

```diff
@@ -224,5 +224,11 @@ class MarshallingDirectivesSpec extends RoutingSpec with Inside {
         rejection shouldEqual UnacceptedResponseContentTypeRejection(Set(ContentType(`application/json`)))
       }
     }
+    "render JSON response when `Accept` header is present with charset UTF-8 explicitly specified" in {
```

Comment:
I think we need more tests for more combination of `Accept` / `Accept-Charset` parameters to be sure it works with all combinations.